packages/flutter/lib/src/material/switch.dart

```diff
@@ -343,6 +344,10 @@ class _SwitchState extends State<Switch> with TickerProviderStateMixin {
       child: Builder(
         builder: (BuildContext context) {
           return _SwitchRenderObjectWidget(
+            // FocusableActionDetector changes the shape of the tree when the
```

Comment:
Should we instead fix `FocusableActionDetector` to not remove the `Actions` and `Shortcuts` widgets when disabled, and instead just empty out their maps? There is some performance cost to building them, but if they cause churn in the tree, that could be offset by not causing it anymore.

Reply:
It would break some tests because that introduces a 1 frame delay to descendant `Focus`'s semantics (`isFocusable`) . Currently since we change the shape of the tree, everything a new `Focus` widget gets inflated so there's no delay (line 1090):

https://github.com/flutter/flutter/blob/4195ad4ef605851be64e649ec2d292645d7425f7/packages/flutter/lib/src/widgets/actions.dart#L1081-L1103

Reply:
(Is this expected?)

Reply:
Well, I would think that anything that cares about the focus changing shouldn't care about the 1 frame delay, since focus changes already happen in a microtask, so that already happens often.  And semantics shouldn't care about a 1 frame delay either.

Reply:
I think I added that part because the focus changes were increasing a build benchmark, and I was trying to reduce the number of built items for things that were disabled.

Reply:
If a switch is disabled then `isFocusable` should be removed from its semantics flags in the same frame, no?

Reply:
The focus itself won't actually be removed until the microtask runs, so if the semantics removes it immediately, it might actually be early (and wrong until the task runs).